renku/core/management/datasets.py

```diff
@@ -469,21 +470,31 @@ def _add_from_urls(
         return files
 
     def _add_from_url(self, dataset, url, destination, extract, progress=None):
-        """Process an add from url and return the location on disk."""
+        """Process add from url and return the location on disk."""
         if destination.exists() and destination.is_dir():
             u = parse.urlparse(url)
             destination = destination / Path(u.path).name
         else:
             destination.parent.mkdir(parents=True, exist_ok=True)
 
         try:
+            start = time.time() * 1e+3
             paths = _download(
                 url=url,
                 download_to=destination,
                 extract=extract,
                 progress_class=progress
             )
-        except error.HTTPError as e:  # pragma nocover
+
+            exec_time = (time.time() * 1e+3 - start) // 1e+3
```

Comment:
I am not sure these multiplication and division by 1000 is needed. We can use the output of `time.time()` and then check if `exec_time < 1` to have the same effect and it's easier to understand the logic.